addons/ai_apps/models/discuss_channel.py

```diff
@@ -0,0 +1,208 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import json
+import datetime
+import pytz
+from markupsafe import Markup
+from markdown import markdown
+
+from odoo import fields, models, api, Command
+from odoo.exceptions import AccessError
+
+from odoo.addons.iap.tools import iap_tools
+from odoo.tools.mail import html_sanitize
+
+
+DEFAULT_OLG_ENDPOINT = 'https://olg.api.odoo.com'
+
+class DiscussChannel(models.Model):
+    """ Chat Session
+        Reprensenting a conversation between users.
+        It extends the base method for usage with AI assistant.
+    """
+
+    _name = 'discuss.channel'
```

Comment:
(up to you for this module, but I'd suggest applying ruff formatter especially in new files)